lib/generate/Request/analyze.js

```diff
@@ -48,7 +48,9 @@ function address(address, feature, result) {
   let urijsPath = './libs/shim/urijs.js';
   if (aid.variableStart.test(addressText)) {
     // Runtime scheme processing necessary
-    feature.address = new URI(addressText);
+    const addressURI = new URI(addressText);
```

Comment:
I haven't tested this, so the syntax might not be 100%, but I think it would make more sense to just run it once, regardless of the path chosen, in the end of the function. What do you think?

```diff
function address(address, feature, result) {
  const addressText = address.toString();
  let urijsPath = './libs/shim/urijs.js';
  if (aid.variableStart.test(addressText)) {
    // Runtime scheme processing necessary
- const addressURI = new URI(addressText);
- const addressQuery = addressURI.query().replace(/\s/g, '%20');
- feature.address = addressURI.query(addressQuery);
+ feature.address = new URI(addressText);
    // Handle urijs import for separate option
    if (
      result.setting &&
      result.setting.separate &&
      result.setting.separate === true
    ) {
      urijsPath = '../../libs/shim/urijs.js';
    }
    result.effectImports.add(urijsPath);
  } else if (aid.schemeStart.test(addressText)) {
    // Scheme specified
    feature.address = new URI(addressText);
  } else {
    // Scheme missing
    feature.address = new URI(`http://${addressText}`);
  }
+ feature.address = feature.address.query(URI.encode(feature.address.query()));
}
```

Reply:
I think this is more compact, I hope the URI encode will only modify the "space" and not transform & or + which are valid query characters.

I'll quickly test and give some feedback on this

Reply:
The URI encode too much saddly:

with URI.encode, it generates an address: `{{baseUrl}}/platform/v1/operations?%24orderby%3DlastAction%20asc%26%24top%3D100%26%24skip%3D0%26%24count%3Dfalse`
```
postman[Symbol.for("define")]({
  name: "Get the list of long running API operations",
  id: "88eb6072-0300-4d8f-8c5d-8725aa8d91dd",
  method: "GET",
  address:
    "{{baseUrl}}/platform/v1/operations?%24orderby%3DlastAction%20asc%26%24top%3D100%26%24skip%3D0%26%24count%3Dfalse",
  post(response) {
    // Validate status 2xx
    pm.test("[GET] /platform/v1/operations - Status code is 2xx", function() {
      pm.response.to.be.success;
    });
  }
});
```

while replacing `feature.address = feature.address.query(URI.encode(feature.address.query()));` with `  feature.address = feature.address.query(feature.address.query().replace(/\s/g, '%20'));`

results in an address: `{{baseUrl}}/platform/v1/operations?$orderby=lastAction%20asc&$top=100&$skip=0&$count=false` which is the same as Postman does (see screenshot above)

```
postman[Symbol.for("define")]({
  name: "Get the list of long running API operations",
  id: "b2e62cb6-969b-41e0-9a5f-8727e79e7e65",
  method: "GET",
  address:
    "{{baseUrl}}/platform/v1/operations?$orderby=lastAction%20asc&$top=100&$skip=0&$count=false",
  post(response) {
    // Validate status 2xx
    pm.test("[GET] /platform/v1/operations - Status code is 2xx", function() {
      pm.response.to.be.success;
    });
  }
});

```

Reply:
Awesome! Seems like there's a method for that as well in urijs however :)

https://medialize.github.io/URI.js/docs.html#static-encodeReserved

Reply:
I've implemented and tested it, like this `feature.address = feature.address.query(URI.encodeReserved(feature.address.query()));` and the result is as expected `{{baseUrl}}/platform/v1/operations?$orderby=lastAction%20asc&$top=100&$skip=0&$count=false`

Reply:
PR is updated.